### ISSUE TITLE
Rector memory allocation problem and "empty" console spam

### ISSUE THREAD

**Issue by kaahulkkonen** [rectorphp/rector#1378]
Hello,

We ran in to this issue while updating from Symfony 2.8 to 3.0.

Issue:
When running rector against some of our bundles, rector would start allocating up to 12gb of RAM and eventually crash. Sometimes there would be "empty" console spam, which would crash the terminal.

After some frustration, we found this was happening at the PHPDocParser:
```In TokenIterator.php line 177:

  [PHPStan\PhpDocParser\Parser\ParserException]
  Unexpected token "", expected '*/' at offset 161


Exception trace:
 () at rector/vendor/phpstan/phpdoc-parser/src/Parser/TokenIterator.php:177
 PHPStan\PhpDocParser\Parser\TokenIterator->throwError() at rector/vendor/phpstan/phpdoc-parser/src/Parser/TokenIterator.php:74
 PHPStan\PhpDocParser\Parser\TokenIterator->consumeTokenType() at rector/vendor/rector/rector/packages/BetterPhpDocParser/src/PhpDocParser/BetterPhpDocParser.php:85
 Rector\BetterPhpDocParser\PhpDocParser\BetterPhpDocParser->parse() at rector/vendor/rector/rector/packages/BetterPhpDocParser/src/PhpDocInfo/PhpDocInfoFactory.php:50
 Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory->createFromNode() at rector/vendor/rector/rector/packages/NodeTypeResolver/src/PhpDoc/NodeAnalyzer/DocBlockManipulator.php:531
 Rector\NodeTypeResolver\PhpDoc\NodeAnalyzer\DocBlockManipulator->createPhpDocInfoFromNode() at rector/vendor/rector/rector/packages/NodeTypeResolver/src/PhpDoc/NodeAnalyzer/DocBlockManipulator.php:166
 Rector\NodeTypeResolver\PhpDoc\NodeAnalyzer\DocBlockManipulator->changeType() at rector/vendor/rector/rector/src/Rector/Class_/RenameClassRector.php:94
 Rector\Rector\Class_\RenameClassRector->refactor() at rector/vendor/rector/rector/src/Rector/AbstractRector.php:95
 Rector\Rector\AbstractRector->enterNode() at rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:200
 PhpParser\NodeTraverser->traverseArray() at rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:91
 PhpParser\NodeTraverser->traverse() at rector/vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php:52
 Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse() at rector/vendor/rector/rector/src/Application/FileProcessor.php:110
 Rector\Application\FileProcessor->refactor() at rector/vendor/rector/rector/src/Application/RectorApplication.php:100
 Rector\Application\RectorApplication->Rector\Application\{closure}() at rector/vendor/rector/rector/src/Application/RectorApplication.php:124
 Rector\Application\RectorApplication->tryCatchWrapper() at rector/vendor/rector/rector/src/Application/RectorApplication.php:101
 Rector\Application\RectorApplication->runOnFileInfos() at rector/vendor/rector/rector/src/Console/Command/ProcessCommand.php:152
 Rector\Console\Command\ProcessCommand->execute() at rector/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at rector/vendor/rector/rector/src/Console/Command/AbstractCommand.php:32
 Rector\Console\Command\AbstractCommand->run() at rector/vendor/symfony/console/Application.php:926
 Symfony\Component\Console\Application->doRunCommand() at rector/vendor/symfony/console/Application.php:269
 Symfony\Component\Console\Application->doRun() at rector/vendor/rector/rector/src/Console/Application.php:73
 Rector\Console\Application->doRun() at rector/vendor/symfony/console/Application.php:145
 Symfony\Component\Console\Application->run() at rector/vendor/rector/rector/bin/rector:24
```

Fixing the issue:
Some of our PHPDoc comments in our source files had comments containing only empty spaces such as:
```
    /**
     * Example doc
     *<whitespace><whitespace>
     * @param string $whatever    example
     * @return integer
    */
```

After removing these extra spaces, and only leaving an empty comment line everything started to work. Like so:
```
    /**
     * Example doc
     *
     * @param string $whatever    example
     * @return integer
    */
```

Issue is resolved for us, but this might be of interest to other people. I guess it was a problem from our end, as PHPDoc validator seems very strict about these whitespaces and as far as I know these empty comment lines with spaces are not allowed.

**Comment by TomasVotruba** [comment]
Hi, thanks for reporting and detailed investigation.

https://github.com/phpstan/phpdoc-parser is used here and it should not care about spaces.

I'm unable to replicate with just any set.
What single rule make it crash?

Also full minimal PHP code would be great, e.g. with method and class

**Comment by TomasVotruba** [comment]
All that happens is the space is removed. I think there is something else that goes wrong

![image](https://user-images.githubusercontent.com/924196/57142848-d584af80-6dbd-11e9-9f64-b17f91456887.png)

**Comment by TomasVotruba** [comment]
- Do you run it from Docker?
- What happens when you run `composer show phpstan/phpdoc-parser` in your application? Any other version than `0.3.3` might be causing the error

**Comment by TomasVotruba** [comment]
Probably resolved with never version 0.5.5+
